### PR TITLE
vue-components: extract ValidationMessage sub component from TextInput

### PR DESCRIPTION
**Pull request by jakobw**
Similar to #209 but for the `FeedbackMessage` component.

Bug: [T265005](https://phabricator.wikimedia.org/T265005)

### CODE REVIEW

File: vue-components/src/components/FeedbackMessage.vue [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		/>
+		{{ message }}
+	</div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import Icon from './Icon.vue';
+
+export default Vue.extend( {
```
Comment (Ladsgroup) [inline vue-components/src/components/FeedbackMessage.vue:17]: You had a similar message for Input component saying it's internal and should not be used outside. Maybe we can have that in here too?

**Comment by Ladsgroup** [comment]
One nitpick and good to go.

File: vue-components/src/components/FeedbackMessage.vue (hunk 1 above)
Comment (jakobw) [reply vue-components/src/components/FeedbackMessage.vue:21]: Good point, thanks. Done!

**Review by Ladsgroup — APPROVED** [review]

**Comment by SaiSan-WMDE** [comment]
Leaving a heads-up here that, since the inline feedbackMessage is its own component now, it should have its own page in Storybook, close to the "block messages" (coming soon). Suggested structure:

Message
- Block
- Inline

Would it be worth reconsidering the naming of the "feedbackMessage" component to "inlineMessage", in order to differentaite both variants (they both provide feedback)? This terminology is also taken straight from the WMF Style guide's [Messages page](https://design.wikimedia.org/style-guide/components/messages.html).

**Comment by jakobw** [comment]
Added to storybook, and renamed to ValidationMessage after discussing with @SaiSan-WMDE. Ready for chromatic review!

**Comment by jakobw** [comment]
Hi @SaiSan-WMDE thanks for the comments! Replying inline:


> * I don't think we'll need to nest any pages under Basic and TextInput. Would it be possible to remove the "All" level?

_\*dies inside a little\*_

I expected this to come up and unfortunately I don't know how we can easily get rid of this extra nesting level. If you look closely you'll see that all component stories follow this format of _Component Name_/_some attribute_ (Button/normal, Icon/all, ...). 

The [docs](https://storybook.js.org/docs/vue/api/csf) say "Every component story file consists of a required default export and one or more named exports.", with the default export being tied to the component (top-level navigation item), and the named export being this nesting level that we want to get rid of.

While typing this reply I asked in the DS channel and the other devs also don't see a way to do it with our current setup. I created https://phabricator.wikimedia.org/T265454 and hope we can keep it as is for now :/

> * Should the text of ValidationMessage display styles? I unfortunately think there're some alignment issues too (҂◡_◡)

OMG yes. Sorry. Fixed.

> * I assumed that ValidationMessage would be nested within a "Message" section that will also contain the BlockMessage component. Would that make sense?

Yup, done!

> * I didn't expect to find `wikit-Input--warning` and `--error` in the Basic input page, but I guess these component states are reusable too?

Yep!

**Comment by SaiSan-WMDE** [comment]
> While typing this reply I asked in the DS channel and the other devs also don't see a way to do it with our current setup. I created https://phabricator.wikimedia.org/T265454 and hope we can keep it as is for now :/

Hey @jakobw! Didn't mean to make you die inside 😭 I thought that this could be either solved in one second or require to change the whole page set up. Thanks so much for creating that ticket! We definitely can leave it as is 👍 

Thanks for all the fixes & changes 🙇‍♀️  All looks 💯 now!

**Pull request merged**